runners/core-construction-java/src/main/java/org/apache/beam/runners/core/construction/graph/GreedyPipelineFuser.java

```diff
@@ -186,14 +185,6 @@ private DescendantConsumers getRootConsumers(PTransformNode rootNode) {
         "Transform %s is not at the root of the graph (consumes %s)",
         rootNode.getId(),
         rootNode.getTransform().getInputsMap());
-    checkArgument(
```

Comment:
Why remove this? Isn't READ_TRANSFORM_URN a runner implemented primitive that should be seen as such?

Reply:
That should not be necessary if the transform is declared as native transform: https://github.com/apache/beam/blob/4adb728535719fa93a775f03dc9023556f008e0f/runners/flink/src/main/java/org/apache/beam/runners/flink/FlinkStreamingPortablePipelineTranslator.java#L439

Reply:
We can't declare it as native transform because we cannot natively translate Reads containing Python code. I suppose we could check if contains a Java payload in `IsNativeTransform`.

Reply:
That or should we use a different URN altogether?

Reply:
The problem with that is that it breaks the expansion logic. We return a `Read` from `GenerateSequence` which during pipeline proto creation gets mapped to the Read URN. See `ReadTranslation`. We would have to change something there to generate a different URN.

Reply:
I've re-added and extended the check.